### PR TITLE
docs(api): new page on adapting OT-2 protocols for Flex

### PR DESCRIPTION
**Pull request by ecormany**
# Overview

New page of the PAPI docs, explaining the minimal steps to turn an OT-2 protocol into a Flex protocol. Based on some similar material I wrote for the Flex manual, but adjusted to fit into these docs, including cross-references to other pages, methods, etc.

# Test Plan

- Check prose and formatting in the [sandbox](http://sandbox.docs.opentrons.com/docs-adapt-ot2-protocols/v2/adapting_ot2_flex.html)
- Check that all standalone code snippets pass analysis

# Changelog

- Brand new page, Adapting OT-2 Protocols for Flex
- Insert new page in the TOC

# Review requests

👍 ❓ 

# Risk assessment

None, docs.

### CODE REVIEW

File: api/docs/v2/adapting_ot2_flex.rst [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+================================
+
+Flex uses different types of pipettes and tip racks than OT-2, which have their own load names in the API. Choose pipettes of the same capacity or larger (or whatever you’ve outfitted your Flex with). See the :ref:`list of pipette API load names <new-pipette-models>` for the valid values of ``instrument_name`` in Flex protocols. And check `Labware Library <https://labware.opentrons.com>`_ or the Opentrons App for the list of Flex tip racks.
+
+This example converts OT-2 code that uses a P300 Single-Channel GEN2 pipette and 300 µL tips to Flex code that uses a Flex 1-Channel 1000 µL pipette and 1000 µL tips.
+
+.. TK code tabs here
+
+
+Deck Slot Names
```
Comment (jwwojak) [inline api/docs/v2/adapting_ot2_flex.rst]: Suggest a different section title:

- Deck Slot Coordinates

These aren't really names like calling a deck slot "John" or "Susan."  I'd recommend words that indicate location or position. "Name" doesn't really do that.

File: api/docs/v2/adapting_ot2_flex.rst [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+===============
+
+It's good practice to update numeric labels for :ref:`deck-slots` (which match the labels on an OT-2) to coordinate ones (which match the labels on Flex). This is an optional step, since the two formats are interchangeable.
```
Comment (jwwojak) [inline api/docs/v2/adapting_ot2_flex.rst:90]: Hey, Ed, I think the deck slot changes are extremely important (maybe that's just me?). Could we consider pumping this section up some more, put some stress on these changes? I realize there's a link to the deck slot doc, but it doesn't hurt to add more emphasis. Here's some suggestions:

- Quickly summarize the changes/differences between Flex and OT-2 (repeat the deck slot doc text)
- Keep the link to the deck slots doc in here somewhere.
- Reuse the image from the deck slots doc.

**Possible revision**
> Flex uses a 2-character, alphanumeric coordinate system to identify deck locations. Labels start at slot A1 (back left) through D3 (front right). This system is like the letter-number coordinates used by most standard well plates you may be familiar with.
>
> The OT-2 uses a simple numeric deck slot labeling system to identify deck locations. Labels start at slot 1 (front left) through slot 11 (back center).

PLACEHOLDER FOR REUSED IMAGE HERE 

> When adapting your older protocols for Flex, it’s a good practice to update the numeric OT-2 deck slot locations to match the Flex coordinate system. The two formats are interchangeable, so a protocol that uses `location=1` will work on a Flex. Our API is smart enough to translate between these different systems. However, we strongly encourage you to update and revise your protocols to use the Flex deck slot coordinates. See `:ref: deck-slots` for more information.

File: api/docs/v2/adapting_ot2_flex.rst [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+.. _adapting-ot2-protocols:
+
+********************************
+Adapting OT-2 Protocols for Flex
+********************************
+
+Python protocols designed to run on the OT-2 can't be directly run on Flex without some modifications. This page describes the minimal steps that you need to take to get OT-2 protocols analyzing and running on Flex.
+
+Adapting a protocol for Flex lets you have parity across different Opentrons robots in your lab, or you can extend older protocols to take advantage of new features only available on Flex. Depending on your application, you may need to do additional verification of your adapted protocol.
```
Comment (jwwojak) [inline api/docs/v2/adapting_ot2_flex.rst:11]: That 1st sentence reads a little long. Let's try getting 2 shorter sentences out of it.

> Adapting a protocol for Flex lets you have parity across different Opentrons robots in your lab. Also, updating an older protocol lets you take advantage of new features only available on Flex. Depending on your application, you may need to do additional verification of your updated code.

File: api/docs/v2/adapting_ot2_flex.rst [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    Consult the list of :ref:`version-notes` to see what effect raising the ``apiLevel`` will have. If you increased it by multiple minor versions to get your protocol running on Flex, make sure that your protocol isn't using removed commands or commands whose behavior has changed in a way that may affect your scientific results.
+
+You also need to specify ``'robotType': 'Flex'``. If you omit ``robotType`` in the ``requirements`` dictionary, the API will assume the protocol is designed for the OT-2.
+
+.. TK code tabs here
+
+Pipettes and Tip-rack Load Names
+================================
+
+Flex uses different types of pipettes and tip racks than OT-2, which have their own load names in the API. Choose pipettes of the same capacity or larger (or whatever you’ve outfitted your Flex with). See the :ref:`list of pipette API load names <new-pipette-models>` for the valid values of ``instrument_name`` in Flex protocols. And check `Labware Library <https://labware.opentrons.com>`_ or the Opentrons App for the list of Flex tip racks.
```
Comment (jwwojak) [inline api/docs/v2/adapting_ot2_flex.rst]: The first sentence here (line 30), consider this alternative:

> Flex pipettes, tip racks, and `:ref: API load names <new-pipette-models>` are different from their OT-2 counterparts.

For the 2nd sentence, I didn't really understand what it's saying. Perhaps try this:

> When updating your protocols to work with Flex, use Flex pipettes of the same capacity or larger as an OT-2 pipette.

And leave the last sentence with the link to the Labware Library.

File: api/docs/v2/adapting_ot2_flex.rst [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+Module Load Names
+=================
+
+If your OT-2 protocol uses older generations of the Temperature Module or Thermocycler Module, update the load names you pass to :py:meth:`.load_module` to ones compatible with Flex:
+
+    * ``temperature module gen2``
+    * ``thermocycler module gen2`` or ``thermocyclerModuleV2``
+    
+The Heater-Shaker Module only has one generation, which is compatible with Flex and OT-2.
```
Comment (jwwojak) [inline api/docs/v2/adapting_ot2_flex.rst]: As the text lists the other load names, consider working the Heater-Shaker load name into this sentence too.

File: api/docs/v2/adapting_ot2_flex.rst [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+The Magnetic Module is not compatible with Flex. For protocols that load ``magnetic module``, ``magdeck``, or ``magnetic module gen2``, you will need to make further modifications to use the :ref:`magnetic-block` and Flex Gripper instead. This will require reworking some of your protocol steps, and you should verify that your new protocol design achieves similar results.
```
Comment (jwwojak) [inline api/docs/v2/adapting_ot2_flex.rst:105]: Boom! Super solid. Love it.

**Review by jwwojak — APPROVED** [review]
@ecormany  left a few comments for your consideration. In the interest of keeping the momentum, I'll add my approval and leave any changes to your discretion.

File: api/docs/v2/adapting_ot2_flex.rst (hunk 1 above)
Comment (ecormany) [reply api/docs/v2/adapting_ot2_flex.rst]: using "Labels" to match what's in the Deck Slots article.

File: api/docs/v2/adapting_ot2_flex.rst (hunk 2 above)
Comment (ecormany) [reply api/docs/v2/adapting_ot2_flex.rst:90]: The deck slot changes may be important for readability, but are completely immaterial to the API. The API does not care. The protocol will analyze and run. Hence the lower emphasis here.

File: api/docs/v2/adapting_ot2_flex.rst (hunk 4 above)
Comment (ecormany) [reply api/docs/v2/adapting_ot2_flex.rst]: agreed on sentence 2, making some updates there.

**Pull request merged**